dags/property_definitions.py

```diff
@@ -1,273 +1,299 @@
-        end_formatted = format_datetime_for_clickhouse(hour_end)
-        time_filter = f"timestamp BETWEEN toDateTime('{start_formatted}') AND toDateTime('{end_formatted}')"
-    else:
-        # Default to previous complete hour
-        now = datetime.datetime.now(datetime.UTC)
-        previous_hour = now.replace(minute=0, second=0, microsecond=0) - datetime.timedelta(hours=1)
-        hour_end = previous_hour + datetime.timedelta(hours=1)
-        start_formatted = format_datetime_for_clickhouse(previous_hour)
-        end_formatted = format_datetime_for_clickhouse(hour_end)
-        time_filter = f"timestamp BETWEEN toDateTime('{start_formatted}') AND toDateTime('{end_formatted}')"
-
     # Log the execution parameters
-    context.log.info(
-        f"Ingesting event properties with target_hour={config.target_hour}, "
-        f"start_date={config.start_date}, end_date={config.end_date}, "
-        f"time_filter={time_filter}"
-    )
+    context.log.info(f"Ingesting person properties for {time_range!r}")
 
     # Query to insert event properties into property_definitions table
-    query = f"""
-    INSERT INTO property_definitions (* EXCEPT(version))
+    insert_query = f"""
+    INSERT INTO property_definitions
     SELECT
         team_id,
         team_id as project_id,
-        (arrayJoin(JSONExtractKeysAndValuesRaw(properties)) as x).1 as name,
-        map(
-            34, 'String',
-            98, 'Boolean',
-            100, 'Numeric',
-            105, 'Numeric',
-            117, 'Numeric',
-            0, NULL
-        )[JSONType(x.2)] as property_type,
-        event,
+        (arrayJoin({DetectPropertyTypeExpression('properties')}) as property).1 as name,
+        property.2 as property_type,
+        replaceAll(event, '\\0', '\ufffd') as event,  -- https://github.com/PostHog/posthog/blob/052f4ea40c5043909115f835f09445e18dd9727c/rust/property-defs-rs/src/types.rs#L172
         NULL as group_type_index,
-        1 as type,
+        {int(PropertyDefinition.Type.EVENT)} as type,
+        -- NOTE: not floored, need to check if needed https://github.com/PostHog/posthog/blob/052f4ea40c5043909115f835f09445e18dd9727c/rust/property-defs-rs/src/types.rs#L175
         max(timestamp) as last_seen_at
     FROM events_recent
-    WHERE {time_filter}
+    WHERE
+        {time_range.get_expression("timestamp")}
+        -- https://github.com/PostHog/posthog/blob/052f4ea40c5043909115f835f09445e18dd9727c/rust/property-defs-rs/src/types.rs#L13-L14C52
+        AND event NOT IN ('$$plugin_metrics')
+        -- https://github.com/PostHog/posthog/blob/052f4ea40c5043909115f835f09445e18dd9727c/rust/property-defs-rs/src/types.rs#L187-L191
+        AND length(event) <= 200
+        AND {COMMON_PROPERTY_FILTERS}
     GROUP BY team_id, event, name, property_type
     ORDER BY team_id, event, name, property_type NULLS LAST
     LIMIT 1 by team_id, event, name
-    SETTINGS max_execution_time = {config.max_execution_time}
     """
```

Comment:
This (and other similar SQL queries below) will likely be refactored out so they can be reused on the read path, if not replaced my materialized views entirely